src/project.coffee

```diff
@@ -30,8 +30,8 @@ class Project extends Model
     @consumeServices(packageManager)
 
   destroyed: ->
-    buffer.destroy() for buffer in @buffers
-    repository?.destroy() for repository in @repositories
+    buffer.destroy() for buffer in @buffers.slice()
+    repository?.destroy() for repository in @repositories.slice()
```

Comment:
The existential operator is because the `@repositories` array can contain `null` entries, since its indices correspond to entries in the `directories` array, which can contain non-repository directories.

:+1: That makes sense about an event listener removing a repository or directory.